lib/puppet/type/file/content.rb

```diff
@@ -220,6 +221,19 @@ def get_from_source(source_or_content, &block)
       end
     end
 
+    def get_from_http_source(source, &block)
```

Comment:
I feel like I'd rather see this go through the indirector and avoid having the file type require special knowledge of how files are gotten. What prevents that from working here?